spec.html

```diff
@@ -822,30 +854,41 @@ <h1>FromElementDescriptor ( _element_ )</h1>
     </emu-clause>
 
     <emu-clause id="sec-to-element-descriptors" aoid=ToElementDescriptors>
-      <h1>ToElementDescriptors ( _elementObjects_ )</h1>
+      <h1>ToElementDescriptors ( _elementObjects_ [ , _internalElements_ ] )</h1>
       <emu-alg>
         1. Assert: _elementObject_ is an ECMAScript language value.
         1. If _elementObjects_ is *undefined*, return *undefined*.
         1. Let _elements_ be a new empty List.
         1. Let _elementObjectsList_ be ? IterableToList(_elementObjects_).
         1. For each _elementObject_ in _elementObjectsList_, do
-          1. Let _element_ be ? ToElementDescriptor(_elementObject_).
+          1. Let _element_ be ? ToElementDescriptor(_elementObject_, _internalElements_).
           1. Let _finisher_ be ? Get(_elementObject_, `"finisher"`).
           1. If _finisher_ is not *undefined*, throw a *TypeError* exception.
           1. Let _extras_ be ? Get(_elementObject_, `"extras"`).
           1. If _extras_ is not *undefined*, throw a *TypeError* exception.
           1. Append _element_ to _elements_.
+        1. If _internalElements_ is set and it is not empty, throw a TypeError.
         1. Return _elements_.
       </emu-alg>
+
+      <emu-note>
+        Internal elements are removed from the _internalElements_ Set as they are found in the _elementObjectList_ list. If after iterating over every _elementObject_ it still isn't empty, it is because some internal elements have been removed.
+      </emu-note>
     </emu-clause>
 
     <emu-clause id=sec-to-element-descriptor aoid=ToElementDescriptor>
-      <h1>ToElementDescriptor ( _elementObject_ )</h1>
+      <h1>ToElementDescriptor ( _elementObject_ [, _internalElements_ ] )</h1>
       <p>With parameter _elementObject_, returns an ElementDescriptor.</p>
       <emu-alg>
         1. Assert: _elementObject_ is an ECMAScript language value.
         1. Let _kind_ be ? ToString(? Get(_elementObject_, `"kind"`)).
-        1. If _kind_ is not one of `"initializer"`, `"method"`, or `"field"`, throw a *TypeError* exception.
+        1. If _kind_ is not one of `"initializer"`, `"method"`, `"field"`, or `"internal"`, throw a *TypeError* exception.
+        1. If _kind_ is `"internal"`,
+          1. If _internalElements_ is not set, throw a TypeError.
+          1. If _elementObject_ doesn't have an [[Element]] internal slot, throw a TypeError.
+          1. If _internalElements_ doesn't contain _elementObject_.[[Element]], throw a TypeError.
```

Comment:
I kept these three checks separate because they are conceptually three different things:
1. `{ kind: "internal" }` was found in a descriptor outside of a class descriptor.
1. It was directly created by the user and not by the engine
1. It comes from another class.